plaso/engine/artifacts_filter_file.py

```diff
@@ -0,0 +1,272 @@
+          'Unable to read artifact definitions from: {0:s} with '
+          'error: ''{1!s}').format(self._path, exception))
+
+    for definition in artifact_registry.GetDefinitions():
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(path_entry, source.separator,
+                                                path_attributes, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(key_entry,
+                                                      path_attributes,
+                                                      find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
+          logging.warning(('Unable to handle Registry Value, extracting '
+                           'key only: {0:s} ').format(source.key_value_pairs))
+
+          for key_pair in source.key_value_pairs:
+            keys = set()
+            keys.add(key_pair.get('key'))
+            for key_entry in keys:
+              if self._CheckKeyCompatibility(key_entry):
+                self.BuildFindSpecsFromRegistryArtifact(key_entry,
+                                                        path_attributes,
+                                                        find_specs)
+        else:
+          logging.warning(('Unable to handle artifact, plaso does not '
+                           'support: {0:s} ').format(source.type_indicator))
+    self._knowledge_base.SetValue(ARTIFACTS_FILTER_FILE, find_specs)
+
+  def BuildFindSpecsFromFileArtifact(self, path_entry, separator,
+                                     path_attributes, find_specs):
+    """Build find specification from a forensic artifacts file.
+
+    Args:
+      path_entry (str):  Current file system path to add.
+          environment variables.
+      separator (str): File system path separator.
+      path_attributes list(str):  Environment variable attributes used to
+          dynamically populate environment variables in key.
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for path in self._ExpandGlobs(path_entry):
+      if path_attributes:
+        try:
+          if '%%environ_' in path:
+            path = path.replace('%%environ_', '{')
+            path = path.replace('%%', '}')
+            path = path.format(**path_attributes)
+        except KeyError as exception:
+          logging.error((
+              'Unable to expand path filter: {0:s} with error: '
+              '{1:s}').format(path, exception))
```

Comment:
use {1!s} to 'cast' exception to a string

Reply:
Done